Observer/Customer/RegisterSuccess.php

```diff
@@ -39,8 +39,24 @@ public function execute(
         if ($this->_smsHelper->getNewCustomerSmsEnabled()) {
             $customer = $observer->getEvent()->getCustomer();
             $this->_logger->info('Customer:', [$customer->getFirstName()]);
-            if (method_exists($customer, 'getBillingAddress')) {
-                $destination = $customer->getBillingAddress()->getTelephone();
+
+            if ($customer instanceof \Magento\Customer\Api\Data\CustomerInterface) {
+                $addresses = $customer->getAddresses();
+
+                foreach ($addresses as $address) {
+                    if ($address->isDefaultBilling()) {
+                        $billingAddress = $address;
+                    }
+                }
+            }
+
+            if (isset($billingAddress) && $billingAddress instanceof \Magento\Customer\Model\Data\Address) {
+                $destination = $billingAddress->getTelephone();
+
+                if (empty($destination)) {
+                    $this->_logger->info('Customer does not have phone number associated to his billing address #' . $billingAddress->getId());
```

Comment:
Does `_logger` need to be renamed to `logger` to keep it consistent with the change made in `Helper/Sms.php`?